apps/admin_audit/lib/Actions/Sharing.php

```
@@ -44,136 +44,168 @@ class Sharing extends Action {
-			);
-		} elseif ($params['shareType'] === IShare::TYPE_REMOTE) {
-			$this->log(
-				'The %s "%s" with ID "%s" has been shared to the remote user "%s" with permissions "%s" (Share ID: %s)',
-				$params,
-				[
-					'itemType',
-					'path',
-					'itemSource',
-					'shareWith',
-					'permissions',
-					'id',
-				]
-			);
-		} elseif ($params['shareType'] === IShare::TYPE_REMOTE_GROUP) {
-			$this->log(
-				'The %s "%s" with ID "%s" has been shared to the remote group "%s" with permissions "%s" (Share ID: %s)',
-				$params,
-				[
-					'itemType',
-					'path',
-					'itemSource',
-					'shareWith',
-					'permissions',
-					'id',
-				]
-			);
-		} elseif ($params['shareType'] === IShare::TYPE_DECK) {
-			$this->log(
-				'The %s "%s" with ID "%s" has been shared to the deck card "%s" with permissions "%s" (Share ID: %s)',
-				$params,
-				[
-					'itemType',
-					'path',
-					'itemSource',
-					'shareWith',
-					'permissions',
-					'id',
-				]
-			);
-		} elseif ($params['shareType'] === IShare::TYPE_SCIENCEMESH) {
-			$this->log(
-				'The %s "%s" with ID "%s" has been shared to the ScienceMesh user "%s" with permissions "%s" (Share ID: %s)',
-				$params,
-				[
-					'itemType',
-					'path',
-					'itemSource',
-					'shareWith',
-					'permissions',
-					'id',
-				]
-			);
-		}
+		match ($params['shareType']) {
+			IShare::TYPE_LINK => $this->sharedLinkType($params),
+			IShare::TYPE_USER => $this->sharedUserType($params),
+			IShare::TYPE_GROUP => $this->sharedGroupType($params),
+			IShare::TYPE_ROOM => $this->sharedRoomType($params),
+			IShare::TYPE_EMAIL => $this->sharedEmailType($params),
+			IShare::TYPE_CIRCLE => $this->sharedCircleType($params),
+			IShare::TYPE_REMOTE => $this->sharedRemoteType($params),
+			IShare::TYPE_REMOTE_GROUP => $this->sharedRemoteGroupType($params),
+			IShare::TYPE_DECK => $this->sharedDeckType($params),
+			IShare::TYPE_SCIENCEMESH => $this->sharedSciencemeshType($params),
+			default => null,
```

I wonder if this one should fail (or at least log) if an unknown type is used.

---

Same here, maybe it will ring a bell for someone :)